index.js

```diff
@@ -11,6 +11,7 @@ function CompressionPlugin(options) {
 	options = options || {};
 	this.asset = options.asset || "[path].gz[query]";
 	this.algorithm = options.algorithm || "gzip";
+	this.changeNewFileName = options.changeNewFileName || false;
```

Comment:
See how we did something similar in extract-text-webpack-plugin. Maybe a good solution to mimic.